### ISSUE TITLE
Refactor our handlers and examples for more code reuse

### ISSUE THREAD

**Issue by msaroufim** [pytorch/serve#2181]
After chatting with folks like @hamelsmu and a lot of folks on the core team it's become clear that we have a reusability problem with the way our examples and handlers are setup so this is an attempt to remedy that

So the goal of this RFC is to reduce the time it takes for someone to write their own custom handler and not provide custom handlers they can just out of the box

Some symptoms of the deeper problem are 
1. Our image classifier example can only resize images up to a specific size https://github.com/pytorch/serve/blob/master/ts/torch_handler/image_classifier.py#L21
2. Most of our large model model capabilities are available for a specific huggingface example https://github.com/pytorch/serve/tree/master/examples/Huggingface_Largemodels
3. Our compilation specific code is interspersed into the based handler as opposed to being standalone utils https://github.com/pytorch/serve/blob/master/ts/torch_handler/base_handler.py so much so that it's almost impossible to understand how torchserve works without reading the code for base handler

So the hot take from this RFC is to
1. Delete all handlers except for the base handler 
2. Refactor all non critical functionality from the base handler into standalone utils that can be easily composed into a new handler script
3. For critical features like large model inference or scriptable tokenizers we refactor them into their own utils

I don't want people to read an example and then copy it over, I want them to just type in `from ts.handler.utils import torchcompile, largemodelinference etc...` - that way they have a reusable set of primitives they can leverage in a more scalable way

**Comment by msaroufim** [comment]
cc @agunapal @HamidShojanazeri @mreso since y'all are the most prolific example producers, lemme know if you have any feedback on this since it will impact how your code is reviewed

**Comment by agunapal** [comment]
@msaroufim This is a good idea. Could you please include a code snippet of how this would look like , say for a BERT text classification model

**Comment by mreso** [comment]
@msaroufim Agreed, I was also thinking if composition would work better for us than inheritance.

We could also thinking about adopting a pattern like torch.data's DataPipes:

`
from ts.torch_handler import Handler
from ts.handler.utils import Load, Compile, DecodeImage, ImageResize, ImageCrop, SoftMaxToLabel

init = Load()
init = Compile(init)

preproc = DecodeImage()
preproc = ImageResize(preproc)
preproc = ImageCrop(preproc)

post = SoftMaxToLabel()

handler = Handler()

handler.initialize = init
handler.preprocess = preproc
handler.postprocess = post
`

**Comment by mreso** [comment]
In fact, I needed something like that for the microbatching PoC and so I gave it a shot to create a backward compatible solution. It would probably be not that hard to create one. Its not yet done but its already partially working with reusable initialization + preproc functions. Not pretty but it works. Can add some syntax sugar to avoid users having to create MethodTypes...
https://github.com/pytorch/serve/tree/feature/microbatching_plus_composable_handler

**Comment by msaroufim** [comment]
Feel free to send it over, might require multiple passes anyways
